src/pages/homepage/PostCard/PostCard.jsx

```diff
@@ -1,53 +1,216 @@
-import "./postcard.css"
+import "./postcard.css";
 import { useData } from "../../../context/dataContext";
-import ReactTimeAgo from 'react-time-ago'
+import ReactTimeAgo from "react-time-ago";
+import { useNavigate, useParams } from "react-router-dom";
+import { useAuth } from "../../../context/authContext";
+import {deletePostServiceHandler, deleteCommentFromPostServiceHandler, postLikeToPostServiceHandler, postDislikeToPostServiceHandler} from "../../../services/services"
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
 
+import { useState } from "react";
+
+export const PostCard = ({ data, fromSinglePostPg, postInfo }) => {
+  const { state, dispatch, TimeAgo, formData, setFormData, setSinglePostPageComment, setPostInformation } = useData();
+  
+  const {
+    username,
+    content,
+    updatedAt,
+    createdAt,
+    likes,
+    _id,
+    comments,
+    commentData
+  } = data;
+
+
+
+  const navigate = useNavigate();
+  const { token, user } = useAuth();
+
+  const findIfLiked = likes?.likedBy.some(el=>el?.username === user?.username)
 
-export const PostCard = ({data}) => {
 
-  const {state, dispatch, TimeAgo} = useData()
-  const {username, content, updatedAt, likes, _id} = data
+  const [viewCardOptionModal, setViewCardOptionModal] = useState(false);
 
-  const timeAgo = new TimeAgo('en-US')
+  const timeAgo = new TimeAgo("en-US");
   let formatedDate = new Date(updatedAt);
 
-  let userData = state.allUsers.filter(el => el.username === username)[0]
+  let userData = state.allUsers.filter((el) => el.username === username)[0];
 
+  const navigateToPostFunc = () => {
+    navigate(`/singlepostpage/${username}/${_id}`);
+  };
+
+  const postOptionsHandler = () => {
+    setViewCardOptionModal(!viewCardOptionModal)
+  };
+
+  const editPostHandler = () => {
+    if(!fromSinglePostPg){
+      setFormData(data)
+      dispatch({ type: "TOGGLE_COMMENT_INPUT_MODAL" })
+    }else{
+      setSinglePostPageComment(true)
+      setPostInformation(postInfo)
+      setFormData(data)
+      dispatch({ type: "TOGGLE_COMMENT_INPUT_MODAL" })
+    }
+    setViewCardOptionModal(!viewCardOptionModal)
+  };
+
+  const deletePostHandler = async () => {
+
+    if(!fromSinglePostPg){
+      let delresp = await deletePostServiceHandler({encodedToken:token, postId:data._id})
+      try {
+        if (delresp.status === 200 || delresp.status === 201) {
+          dispatch({ type: "SET_ALL_POSTS", payload: delresp.data.posts });
   
+          toast.error("Post Deleted!", {
+            position: "bottom-right",
+            autoClose: 1500,
+          });
+        }
+      } catch (err) {
+        console.log(err);
```

Comment:
Suggestion : we can show failed error in the ui.

Reply:
I see, didn't think about it that way. Will look into it!